### ISSUE TITLE
Output from MPAS-O unreadable for large 1.8M cell mesh

### ISSUE THREAD

**Issue by pwolfram** [MPAS-Dev/MPAS-Model#576]
Output from initialization step for 1.8M cell mesh appears to produce an incompatible netcdf file:

```
└─▪ ncinfo ocean.nc                                                                                                                                       
Traceback (most recent call last):                                                                                                                        
  File "/usr/projects/climate/SHARED_CLIMATE/anaconda_envs/base/envs/compass_0.1.6/bin/ncinfo", line 11, in <module>                                      
    sys.exit(ncinfo())                                                                                                                                    
  File "/usr/projects/climate/SHARED_CLIMATE/anaconda_envs/base/envs/compass_0.1.6/lib/python3.8/site-packages/netCDF4/utils.py", line 559, in ncinfo     
    f = Dataset(filename)                                                                                                                                 
  File "netCDF4/_netCDF4.pyx", line 2321, in netCDF4._netCDF4.Dataset.__init__                                                                            
  File "netCDF4/_netCDF4.pyx", line 1885, in netCDF4._netCDF4._ensure_nc_success                                                                          
OSError: [Errno -62] NetCDF: One or more variable sizes violate format constraints: b'ocean.nc'                                                           
```

This issue is typically alleviated by setting netcdf file type to be `NETCDF3_64BIT_OFFSET`, which is used for python preprocessing inputs into MPAS-O.  

It is strange there is this error with MPAS-O output.  This precludes use of python-based netcdf tools with the produced MPAS-O output file.

**Comment by pwolfram** [comment]
Thanks to @xylar for some preliminary discussions-- cc'ing @maltrud, @jonbob and @vanroekel in case you've seen this issue with large simulations in the past.

**Comment by pwolfram** [comment]
Stack is intel/17.0.1 openmpi/1.10.5 netcdf/4.4.1 parallel-netcdf/1.5.0 pio/1.7.2

Following discussion with @vanroekel: 

 * there may be a bug in parallel-netcdf/1.5.0
 * option to try may be to set `io_type=pnetcdf,cdf5` in streams

**Comment by vanroekel** [comment]
to follow up, there is definitely a bug in pnetcdf/1.5.0, it allows violation of the netcdf classic size constraint but then leads to corrupt netcdf files.  The files can be read, but only with pnetcdf/1.5.0 and read time is significantly slower

**Comment by pwolfram** [comment]
This means all our standard python tools won't work with the produced files, not to mention it is not performant either.

**Comment by xylar** [comment]
@pwolfram, is the simplest solution perhaps get someone to build the latest pnetcdf (1.12.1) on LANL IC?  1.5.0 is from July 8, 2014, so just a tad out of date...

**Comment by xylar** [comment]
To be clear, I'm not volunteering to take over this job because I feel I do enough "sysadmin/tech support" type work for the group as it is.

**Comment by jonbob** [comment]
@xylar and @pwolfram - if it really is just a question of pnetcdf version, @stephenprice and I have been working with IC to provide module support we need, including pnetcdf. They have added what we need to their spack setup, so we're hoping this support will become more automatic from this point forward. I have a PR that should go on master today that updates grizzly and badger machine files and includes pnetcdf/1.11.2.

**Comment by stephenprice** [comment]
Jon and I have been working with LANL HPC to update our library support there. If you take a look at the latest machine files for Grizzly (@jonbob can you point to that PR / branch if it has not yet been merged?) you can see what's now available. W.r.t. this thread, there's pnetcdf 1.11.2 working there now (for both Griz and Badger).

**Comment by xylar** [comment]
Jinx!

**Comment by stephenprice** [comment]
He beat me by a minute ...

**Comment by jonbob** [comment]
Ha!

**Comment by xylar** [comment]
@pwolfram, can you retry with the modules in: https://github.com/E3SM-Project/E3SM/pull/3554/files
```
module load intel/19.0.4 openmpi/2.1.2  hdf5-parallel/1.8.16 pnetcdf/1.11.2 netcdf-h5parallel/4.7.3
```

That file also lists:
```
friendly-testing mkl/2019.0.4
```
@jonbob, would those also be needed?

Also, E3SM builds PIO whereas MPAS-Ocean doesn't.  Is there a module for that?

**Comment by pwolfram** [comment]
I spent a chunk of yesterday using this stack (thanks to @vanroekel !) to build PIO but wasn't able to get it to work.  Could be doing something slightly wrong, and tried with and without compass environment loaded because sometimes that can cause issues.  Errors I got were like this and occurred on linking:

```
make[3]: Leaving directory `/lustre/scratch4/turquoise/.mdt1/pwolfram/MPAS_add_flooding/src/driver'
mpif90 -O3 -o ocean_model driver/*.o -L. -ldycore -lops -lframework -L/usr/projects/climate/SHARED_CLIMATE/software/grizzly/pio/2.5.0/intel-19.0.4/openmpi-2.1.2/netcdf-4.7.3-parallel-netcdf-1.11.2/lib -lpiof -lpioc -I./external/esmf_time_f90 -L./external/esmf_time_f90 -lesmf_time
/usr/projects/climate/SHARED_CLIMATE/software/grizzly/pio/2.5.0/intel-19.0.4/openmpi-2.1.2/netcdf-4.7.3-parallel-netcdf-1.11.2/lib/libpioc.a(pio_file.c.o): In function `PIOc_closefile':
/usr/projects/climate/SHARED_CLIMATE/software/grizzly/pio/2.5.0/intel-19.0.4/openmpi-2.1.2/ParallelIO/src/clib/pio_file.c:459: undefined reference to `nc_sync'
/usr/projects/climate/SHARED_CLIMATE/software/grizzly/pio/2.5.0/intel-19.0.4/openmpi-2.1.2/ParallelIO/src/clib/pio_file.c:281: undefined reference to `ncmpi_close'
/usr/projects/climate/SH. ......
```

Environment used is : `module load intel/19.0.4 openmpi/2.1.2 hdf5-parallel/1.8.16 pnetcdf/1.11.2 netcdf-h5parallel/4.7.3 mkl/2019.0.4 pio/2.5.0` where `pio/2.5.0` was my own apparently flawed build (everything built fine and appeared to link properly to other libraries above)

**Comment by jonbob** [comment]
@xylar - yes, you would need the "friendly-testing" module, since it gives access to the other new ones. mkl is necessary to compile E3SM, but I don't know about MPAS-Ocean. As for PIO, we did not add a module for that.

**Comment by jonbob** [comment]
@pwolfram - you should be able to build a test case using E3SM next, which has this PR already merged, and just point at that version of PIO

**Comment by pwolfram** [comment]
Command to build PIO, for the record, was 

```
git clone [mailto:git@github.com:PARALLELIO/ParallelIO.git]git@github.com:PARALLELIO/ParallelIO.git
cd ParallelIO (I actually usually rename the dir pio)
mkdir build
cd build
CC=mpicc FC=mpif90 cmake -DCMAKE_INSTALL_PREFIX=/usr/projects/climate/SHARED_CLIMATE/software/grizzly/pio/2.5.0/intel-19.0.4/openmpi-2.1.2/netcdf-4.7.3-parallel-netcdf-1.11.2/ -DPIO_ENABLE_TIMING=OFF -DNetCDF_Fortran_PATH=/usr/projects/hpcsoft/toss3/grizzly/netcdf/4.7.3_intel-19.0.4_openmpi-2.1.2_hdf5-1.8.16/ -DPnetCDF_Fortran_PATH=/usr/projects/hpcsoft/toss3/grizzly/pnetcdf/1.11.2_intel-19.0.4_openmpi-2.1.2_hdf5-1.8.16/ -DNetCDF_PATH=/usr/projects/hpcsoft/toss3/grizzly/netcdf/4.7.3_intel-19.0.4_openmpi-2.1.2_hdf5-1.8.16/ -DPnetCDF_Fortran_PATH=/usr/projects/hpcsoft/toss3/grizzly/pnetcdf/1.11.2_intel-19.0.4_openmpi-2.1.2_hdf5-1.8.16/ -DPnetCDF_C_PATH=/usr/projects/hpcsoft/toss3/grizzly/pnetcdf/1.11.2_intel-19.0.4_openmpi-2.1.2_hdf5-1.8.16/ ..
make
make install
```

**Comment by pwolfram** [comment]
I put this into the `module pio/2.5.0` and then set up environment with
```
module purge
module use /usr/projects/climate/SHARED_CLIMATE/modulefiles/all/
module load friendly-testing
module load intel/19.0.4 openmpi/2.1.2 hdf5-parallel/1.8.16 pnetcdf/1.11.2 netcdf-h5parallel/4.7.3 mkl/2019.0.4 pio/2.5.0
```

However, MPAS-O did not build using `make ifort CORE=ocean DEBUG=false AUTOCLEAN=true` and failed on the PIO testing step.

**Comment by pwolfram** [comment]
@jonbob, how should PIO get built?  Can you please send along a link to your PR as @stephenprice mentions?

**Comment by xylar** [comment]
@pwolfram, I linked to the PR already above.

**Comment by pwolfram** [comment]
Thanks @xylar, got it!  Will try again from there.

**Comment by xylar** [comment]
@pwolfram but @jonbob was suggesting you use the E3SM/next branch.

If you build a test case (any test case) in E3SM, it will build PIO as part of building E3SM.  Then, you can just figure out where and copy that over.

**Comment by jonbob** [comment]
@pwolfram - ideally you'd be building E3SM instead of a standalone model, and then it would magically happen! I don't know about building it on its own -- I never do much outside the system itself

**Comment by jonbob** [comment]
And it is on master now...

**Comment by xylar** [comment]
> ideally you'd be building E3SM instead of a standalone model, and then it would magically happen!

Hmm, clearly definitions of "ideally" can differ ;-)

**Comment by jonbob** [comment]
Ha! Yes, ideally if only for one workflow!

**Comment by pwolfram** [comment]
Is there an easy way to know which version of PIO is being used for E3SM @jonbob?

**Comment by philipwjones** [comment]
E3SM uses SCORPIO - https://github.com/E3SM-Project/scorpio
and you should build that version.

**Comment by pwolfram** [comment]
@philipwjones and @jonbob, do you recommend I build with E3SM and then copy over lib / include into a module file?  

I'm assuming the latest version of scorpio is the one to use or should I use a release version like `scorpio-v1.1.0`?

**Comment by philipwjones** [comment]
I'm guessing a tagged release is safer.  FYI, I've built successfully on summit starting with a pretty basic CMake like (obviously change the target install paths):

CC=mpicc FC=mpif90 cmake -DCMAKE_INSTALL_PREFIX=~/summitlibs/pio -DPIO_ENABLE_TIMING=OFF -
DNetCDF_Fortran_PATH=${OLCF_NETCDF_FORTRAN_ROOT} ~/summitlibs/pio

**Comment by pwolfram** [comment]
@mark-petersen or @xylar, can you please confirm this works for you for a build?

You can set up the environment with 
```
module purge
module load friendly-testing
module load intel/19.0.4 openmpi/2.1.2 hdf5-parallel/1.8.16 pnetcdf/1.11.2 netcdf-h5parallel/4.7.3 mkl/2019.0.4
export PIO=/usr/projects/climate/SHARED_CLIMATE/software/grizzly/pio/1.10.1/intel-19.0.4/openmpi-2.1.2/netcdf-h5parallel-4.7.3-pnetcdf-1.11.2/
export NETCDF=/usr/projects/hpcsoft/toss3/grizzly/pnetcdf/1.11.2_intel-19.0.4_openmpi-2.1.2_hdf5-1.8.16/
export PNETCDF=/usr/projects/hpcsoft/toss3/grizzly/pnetcdf/1.11.2_intel-19.0.4_openmpi-2.1.2_hdf5-1.8.16/
```
I would plan to make this more robust if indeed I can get MPAS-O to work correctly, but have not done this now while in debugging.

The `io_type=pnetcdf,cdf5` option did not work with this build, and is needed for the large ocean.nc stream  and appears to be giving lustre library errors.  Smaller meshes worked as expected with the previous library (without a change to `io_type`).

```
forrtl: severe (71): integer divide by zero
Image              PC                Routine            Line        Source             
ocean_model        00000000031C4864  Unknown               Unknown  Unknown
libpthread-2.17.s  00002B0A7FB3E630  Unknown               Unknown  Unknown
libmpi.so.20.10.2  00002B0A7F4CEA72  ADIOI_LUSTRE_Get_     Unknown  Unknown
libmpi.so.20.10.2  00002B0A7F4D0764  ADIOI_LUSTRE_Writ     Unknown  Unknown
libmpi.so.20.10.2  00002B0A7F4FBD68  MPIOI_File_write_     Unknown  Unknown
libmpi.so.20.10.2  00002B0A7F4FC552  mca_io_romio314_d     Unknown  Unknown
libmpi.so.20.10.2  00002B0A7F46078B  MPI_File_write_at     Unknown  Unknown
libpnetcdf.so.3.0  00002B0A7E30DF82  ncmpio_read_write     Unknown  Unknown
libpnetcdf.so.3.0  00002B0A7E308E83  Unknown               Unknown  Unknown
libpnetcdf.so.3.0  00002B0A7E306639  Unknown               Unknown  Unknown
libpnetcdf.so.3.0  00002B0A7E304235  Unknown               Unknown  Unknown
libpnetcdf.so.3.0  00002B0A7E2411A2  ncmpi_wait_all        Unknown  Unknown
ocean_model        000000000318E0E5  Unknown               Unknown  Unknown
ocean_model        00000000031850E1  Unknown               Unknown  Unknown
ocean_model        000000000318E8B3  Unknown               Unknown  Unknown
ocean_model        0000000003185C22  Unknown               Unknown  Unknown
ocean_model        0000000003135098  Unknown               Unknown  Unknown
ocean_model        0000000002DAC4E5  mpas_io_mp_mpas_i        2473  mpas_io.F
ocean_model        0000000002DD6F99  mpas_io_mp_mpas_i        3032  mpas_io.F
ocean_model        0000000002E85956  mpas_io_streams_m        3605  mpas_io_streams.F
ocean_model        0000000002EDD55A  mpas_stream_manag        3356  mpas_stream_manager.F
ocean_model        0000000002ED9401  mpas_stream_manag        2820  mpas_stream_manager.F
ocean_model        0000000001BE5604  ocn_init_mode_mp_         297  mpas_ocn_init_mode.F
ocean_model        000000000272E69B  ocn_core_mp_ocn_c         115  mpas_ocn_core.F
ocean_model        0000000000417826  mpas_subdriver_mp         347  mpas_subdriver.F
ocean_model        00000000004128DA  MAIN__                     16  mpas.F
ocean_model        0000000000412852  Unknown               Unknown  Unknown
libc-2.17.so       00002B0A7FF71545  __libc_start_main     Unknown  Unknown
ocean_model        0000000000412769  Unknown               Unknown  Unknown
```

I changed pio options from
```
 config_pio_num_iotasks = 0
 config_pio_stride = 1
```
to 
```
 config_pio_num_iotasks = 10
 config_pio_stride = 36
```
for 10 nodes on grizzly but then started getting errors like
```
forrtl: severe (408): fort: (2): Subscript #1 of the array EDGESONCELL has value 8 which is greater than the upper bound of 7

Image              PC                Routine            Line        Source             
ocean_model        00000000031BBAE6  Unknown               Unknown  Unknown
ocean_model        0000000002C3B4A9  mpas_block_decomp         387  mpas_block_decomp.F
ocean_model        0000000003064236  mpas_block_creato         315  mpas_block_creator.F
ocean_model        0000000002EAE7D1  mpas_bootstrappin         251  mpas_bootstrapping.F
ocean_model        0000000000416602  mpas_subdriver_mp         293  mpas_subdriver.F
ocean_model        00000000004128D5  MAIN__                     14  mpas.F
ocean_model        0000000000412852  Unknown               Unknown  Unknown
libc-2.17.so       00002BAA261F9545  __libc_start_main     Unknown  Unknown
ocean_model        0000000000412769  Unknown               Unknown  Unknown
```
and was not able to get this to work based on others' prior experiences (reported to me in past as well as via a github search)...

Has anyone experienced this type of error before?  This is with the `scorpio-v1.1.0` pio 1.10.0 build, noting it has to be compiled with `USE_PIO2=true`.  

What was the largest E3SM run performed with this new stack?  240km cases won't stress test this type of issue.

All advice is appreciated!

**Comment by xylar** [comment]
@pwolfram, I build with your configuration and ran the QU240 test case.  (I didn't build in DEBUG mode, which it looks like you did.)  I get the follow error in the log file with the default setup of `initial_state`:
```
CRITICAL ERROR: Error in block_decomp_partitioned_edge_list: edge/vertex is not adjacent to any valid cells
```

I don't have any experience building PIO2 or `scorpio` so I don't think I can be of much help on this one.  But I'm happy to do some testing once you have something you think might be working as expected.

**Comment by pwolfram** [comment]
@xylar, it is this one `compass/ocean/global_ocean/QU240/init/`, right?  I'm getting the same error as at high-resolution for this case, which appears to be failing in the lustre library calls.  This is for `debug=true`:

```
forrtl: severe (71): integer divide by zero
Image              PC                Routine            Line        Source             
ocean_model        00000000031C4864  Unknown               Unknown  Unknown
libpthread-2.17.s  00002AECC037D630  Unknown               Unknown  Unknown
libmpi.so.20.10.2  00002AECBFD0DA72  ADIOI_LUSTRE_Get_     Unknown  Unknown
libmpi.so.20.10.2  00002AECBFD0F764  ADIOI_LUSTRE_Writ     Unknown  Unknown
libmpi.so.20.10.2  00002AECBFD3AD68  MPIOI_File_write_     Unknown  Unknown
libmpi.so.20.10.2  00002AECBFD3B552  mca_io_romio314_d     Unknown  Unknown
libmpi.so.20.10.2  00002AECBFC9F78B  MPI_File_write_at     Unknown  Unknown
libpnetcdf.so.3.0  00002AECBEB4CF82  ncmpio_read_write     Unknown  Unknown
libpnetcdf.so.3.0  00002AECBEB47E83  Unknown               Unknown  Unknown
libpnetcdf.so.3.0  00002AECBEB45639  Unknown               Unknown  Unknown
libpnetcdf.so.3.0  00002AECBEB43235  Unknown               Unknown  Unknown
libpnetcdf.so.3.0  00002AECBEA801A2  ncmpi_wait_all        Unknown  Unknown
ocean_model        000000000318E0E5  Unknown               Unknown  Unknown
ocean_model        0000000003184CFC  Unknown               Unknown  Unknown
ocean_model        000000000318E8B3  Unknown               Unknown  Unknown
ocean_model        0000000003185C22  Unknown               Unknown  Unknown
ocean_model        000000000313B1B6  Unknown               Unknown  Unknown
ocean_model        0000000002DB18C7  mpas_io_mp_mpas_i        2540  mpas_io.F
ocean_model        0000000002DD7286  mpas_io_mp_mpas_i        3049  mpas_io.F
ocean_model        0000000002E91B7B  mpas_io_streams_m        3733  mpas_io_streams.F
ocean_model        0000000002EDD55A  mpas_stream_manag        3356  mpas_stream_manager.F
ocean_model        0000000002ED9401  mpas_stream_manag        2820  mpas_stream_manager.F
ocean_model        0000000001BE5604  ocn_init_mode_mp_         297  mpas_ocn_init_mode.F
ocean_model        000000000272E69B  ocn_core_mp_ocn_c         115  mpas_ocn_core.F
ocean_model        0000000000417826  mpas_subdriver_mp         347  mpas_subdriver.F
ocean_model        00000000004128DA  MAIN__                     16  mpas.F
ocean_model        0000000000412852  Unknown               Unknown  Unknown
libc-2.17.so       00002AECC07B0545  __libc_start_main     Unknown  Unknown
ocean_model        0000000000412769  Unknown               Unknown  Unknown
```

This is for `debug=false` gives a similar error...

```
forrtl: severe (71): integer divide by zero
Image              PC                Routine            Line        Source             
ocean_model        0000000000C69734  for__signal_handl     Unknown  Unknown
libpthread-2.17.s  00002B7C9F836630  Unknown               Unknown  Unknown
libmpi.so.20.10.2  00002B7C9F1C6A72  ADIOI_LUSTRE_Get_     Unknown  Unknown
libmpi.so.20.10.2  00002B7C9F1C8764  ADIOI_LUSTRE_Writ     Unknown  Unknown
libmpi.so.20.10.2  00002B7C9F1F3D68  MPIOI_File_write_     Unknown  Unknown
libmpi.so.20.10.2  00002B7C9F1F4552  mca_io_romio314_d     Unknown  Unknown
libmpi.so.20.10.2  00002B7C9F15878B  MPI_File_write_at     Unknown  Unknown
libpnetcdf.so.3.0  00002B7C9E005F82  ncmpio_read_write     Unknown  Unknown
libpnetcdf.so.3.0  00002B7C9E000E83  Unknown               Unknown  Unknown
libpnetcdf.so.3.0  00002B7C9DFFE639  Unknown               Unknown  Unknown
libpnetcdf.so.3.0  00002B7C9DFFC235  Unknown               Unknown  Unknown
libpnetcdf.so.3.0  00002B7C9DF391A2  ncmpi_wait_all        Unknown  Unknown
ocean_model        0000000000C3B565  Unknown               Unknown  Unknown
ocean_model        0000000000C3217C  Unknown               Unknown  Unknown
ocean_model        0000000000C3BD33  Unknown               Unknown  Unknown
ocean_model        0000000000C330A2  Unknown               Unknown  Unknown
ocean_model        0000000000BE8636  Unknown               Unknown  Unknown
ocean_model        0000000000A935CE  Unknown               Unknown  Unknown
ocean_model        0000000000A96B70  Unknown               Unknown  Unknown
ocean_model        0000000000AC8C32  Unknown               Unknown  Unknown
ocean_model        0000000000AD9CBE  Unknown               Unknown  Unknown
ocean_model        0000000000AD96B4  Unknown               Unknown  Unknown
ocean_model        0000000000636D4B  Unknown               Unknown  Unknown
ocean_model        000000000090A23F  Unknown               Unknown  Unknown
ocean_model        0000000000412AE7  Unknown               Unknown  Unknown
ocean_model        0000000000412A6E  Unknown               Unknown  Unknown
ocean_model        0000000000412A12  Unknown               Unknown  Unknown
libc-2.17.so       00002B7C9FA65545  __libc_start_main     Unknown  Unknown
ocean_model        0000000000412929  Unknown               Unknown  Unknown

```

**Comment by pwolfram** [comment]
@philipwjones, did you or your team see any `ADIOI_LUSTRE` similar errors in your work?  Is it worth posting this issue to the scorpio issue tracker?

**Comment by philipwjones** [comment]
No, we don't typically use the LANL clusters and most of our performance work is with I/O turned off, unless we're doing specific I/O testing. Feel free to report but I know some of their issues on other machines have been related to trying to link to older netcdf/pnetcdf libs. Not sure what the version compatibility is.

**Comment by pwolfram** [comment]
@xylar, quick note, if you `ldd ocean_model`, is the correct version of NETCDF being pointed to in the executable?  I found this as an error this evening and note there was a typo for the previous module load (updated below):

```
module purge
module load friendly-testing
module load intel/19.0.4 openmpi/2.1.2 hdf5-parallel/1.8.16 pnetcdf/1.11.2 netcdf-h5parallel/4.7.3 mkl/2019.0.4
export PIO=/usr/projects/climate/SHARED_CLIMATE/software/grizzly/pio/1.10.1/intel-19.0.4/openmpi-2.1.2/netcdf-h5parallel-4.7.3-pnetcdf-1.11.2/
export NETCDF=/usr/projects/hpcsoft/toss3/grizzly/netcdf/4.7.3_intel-19.0.4_openmpi-2.1.2_hdf5-1.8.16/
export PNETCDF=/usr/projects/hpcsoft/toss3/grizzly/pnetcdf/1.11.2_intel-19.0.4_openmpi-2.1.2_hdf5-1.8.16/
```

where the library for netcdf is actually in `lib64`, not `lib`, which means it was pointing to the system version.  This didn't fix the problem, however, which best I can tell is a pnetcdf error presumably due to some type of bad API call or pnetcdf install.  This can be confirmed by setting `io_type="netcdf"` for the streams, which then complete properly.   `io_type="netcdf4"` did not work properly.

@philipwjones, is `USE_PIO2=true` needed to compile with `scorpio-v1.1.0`?  I'm suspicious this is the problem, that this really isn't PIO2, when the Makefile appears to think that it is actually PIO2..

**Comment by wkliao** [comment]
```
libmpi.so.20.10.2  00002B7C9F1C6A72  ADIOI_LUSTRE_Get_     Unknown  Unknown
libmpi.so.20.10.2  00002B7C9F1C8764  ADIOI_LUSTRE_Writ     Unknown  Unknown
libmpi.so.20.10.2  00002B7C9F1F3D68  MPIOI_File_write_     Unknown  Unknown
libmpi.so.20.10.2  00002B7C9F1F4552  mca_io_romio314_d     Unknown  Unknown
```
This error message is likely caused by a bug that has already been fixed in MPICH 3.3's ROMIO but OpenMPI's ROMIO has not caught up with it. See an issue created by @dqwu in MPICH repo.

**Comment by pwolfram** [comment]
@xylar, can you please try using

```
module purge
module load friendly-testing
module load intel/19.0.4 intel-mpi/2019.4 hdf5-parallel/1.8.16 pnetcdf/1.11.2 netcdf-h5parallel/4.7.3 mkl/2019.0.4
export PIO=/usr/projects/climate/SHARED_CLIMATE/software/grizzly/pio/1.10.1/intel-19.0.4/intel-mpi-2019.4/netcdf-4.7.3-parallel-netcdf-1.11.2/
export NETCDF=/usr/projects/hpcsoft/toss3/grizzly/netcdf/4.7.3_intel-19.0.4_intel-mpi-2019.4_hdf5-1.8.16/
export PNETCDF=/usr/projects/hpcsoft/toss3/grizzly/pnetcdf/1.11.2_intel-19.0.4_intel-mpi-2019.4_hdf5-1.8.16/                                                                
```

where you'll have to ensure the correct netcdf library version is being obtained (in `lib64` not `lib`) as well as using all intel compilers (add to your Makefile).  See gist at https://gist.github.com/2537e6a32a8fe4d85e32e8af76a0c898 for modified Makefile.  I've been double-checking to make sure dynamic linking is correct via `ldd ocean_model` too.

So far, I believe this should work now that we are not using OpenMPI thanks to @wkliao's very helpful comments!

This works for me now for both the `QU240` as well as 100m hurricane cases.

**Comment by philipwjones** [comment]
Looks like you already have a handle, but just to answer the question above - yes, scorpio uses pio2 interfaces. I'd be strongly in favor of removing that flag from the build, or if we can't remove, at least make pio2 interfaces the default.

**Comment by pwolfram** [comment]
I would support this decision given my experiences this last week @philipwjones !

**Comment by xylar** [comment]
@pwolfram, do you have a good plan for fixing this in a permanent way?  Constantly editing the Makefile is obviously not a great option.  Is it time to get cmake working? ;-)

**Comment by xylar** [comment]
I'll test as soon as I have a chance but probably not today.

**Comment by pwolfram** [comment]
@xylar, @mark-petersen already took some steps in this direction with `src/core_ocean/ocean.cmake`. I agree, our software stack / build capability isn't that robust right now but this is probably best served by a broader discussion, which could occur here (or somewhere else).

**Comment by pwolfram** [comment]
This now works with `io_type=pnetcdf,cdf5` for large datasets and a PIO error message is reported indicating that variable sizes are larger than accommodated by the present `io_type`.

**Comment by pwolfram** [comment]
I'm storing up to date instructions at https://acme-climate.atlassian.net/wiki/spaces/ECG/pages/1054113803/MPAS-Ocean+Initialization+and+Testing

**Comment by pwolfram** [comment]
Closed by #592
